java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -479,18 +512,10 @@ public long latestVersion() {
    * @param version the version to check out
    * @return a new Dataset instance with the specified version checked out
    */
-  public Dataset checkoutVersion(long version) {
-    if (version < 1) {
-      throw new IllegalArgumentException("Version must be greater than 0");
-    }
-
+  public Dataset checkout(long version) {
```

Comment:
Why change this name? The original name seems fine to me

Reply:
If you really want to go with this, then let's follow the breaking change policy and adjust the PR title and description: https://github.com/lancedb/lance/blob/main/release_process.md#breaking-change-policy

To me, I think we can just go with `checkoutVersion`, `checkoutTag`, `checkoutTimestamp`, but I am not super opinionated about this.

btw, if you plan to use long for timestamp, shouldn't you also use long for `private final ZonedDateTime dataTime`?

Reply:
> btw, if you plan to use long for timestamp, shouldn't you also use long for private final ZonedDateTime dataTime?

I'm not quite sure about using timestamp as the parameter. This is only for discussion. I checked rust and python there is no similar api. Using ZonedDateTime is because rust code using DataTime<Utc>

I'm OK with checkoutVersion as well as checkoutTag

Reply:
I see, then let's just stick with ZonedDateTime then, although it's always UTC so a bit redundant but that's fine I guess.